### PR TITLE
5303 raw ptr to smart ptr

### PR DESCRIPTION
**Pull request by konangc87**
|Related issue|
|---|
|#5303|

## Description
Raw pointers to SQLite structures were replaced by smart pointers + deleters.

## Tests

- Compilation without warnings in every supported platform
  - [ ] Linux
  - [ ] Windows
  - [ ] MAC OS X
- [ ] Source installation
- [ ] Package installation
- [ ] Source upgrade
- [ ] Package upgrade
- [ ] Review logs syntax and correct language
- [ ] QA templates contemplate the added capabilities

- Memory tests for Linux
  - [ ] Scan-build report
  - [ ] Coverity
  - [ ] Valgrind (memcheck and descriptor leaks check)
  - [ ] Dr. Memory
  - [ ] AddressSanitizer
- Memory tests for Windows
  - [ ] Scan-build report
  - [ ] Coverity
  - [ ] Dr. Memory
- Memory tests for macOS
  - [ ] Scan-build report
  - [ ] Leaks
  - [ ] AddressSanitizer

- [ ] Retrocompatibility with older Wazuh versions
- [ ] Working on cluster environments
- [ ] Configuration on demand reports new parameters
- [ ] The data flow works as expected (agent-manager-api-app)
- [ ] Added unit tests (for new features)
- [ ] Stress test for affected components

### CODE REVIEW

File: src/dbsync/src/sqlite_wrapper.cpp [hunk 1]
```diff
@@ -13,147 +13,193 @@
...
-
-Statement::Statement(std::shared_ptr<IConnection>& connection, const std::string& query) : m_connection(connection){
-  if(SQLITE_OK != sqlite3_prepare_v2(connection->GetDBInstance(), query.c_str(), -1, &m_stmt, nullptr)) {
-    throw SQLite::exception(601, "cannot instance SQLite stmt.");
-  }
+bool Transaction::rollback()
+{
+    if (!m_rolledBack && !m_commited)
+    {
+        m_connection->execute("ROLLBACK TRANSACTION");
```
Comment (tinachoB) [inline src/dbsync/src/sqlite_wrapper.cpp:99]: shouldn't the "execute" return value needed to be checked before going on?

File: src/dbsync/src/sqlite_wrapper.h [hunk 2]
```diff
@@ -15,94 +15,101 @@
...
+
+        bool commit() override;
+        bool rollback() override;
+    private:
+        std::shared_ptr<IConnection> m_connection;
+        bool m_rolledBack;
+        bool m_commited;
+    };
+
+    class Column : public IColumn {
```
Comment (tinachoB) [inline src/dbsync/src/sqlite_wrapper.h:76]: Please, move the "{" in a new line

File: src/dbsync/src/sqlite_wrapper.h [hunk 3]
```diff
@@ -15,94 +15,101 @@
...
+    public:
+        virtual ~Column() = default;
```
Comment (tinachoB) [inline src/dbsync/src/sqlite_wrapper.h:78]: Is this "virtual" necessary here?

File: src/dbsync/src/sqlite_wrapper.cpp (hunk 1 above)
Comment (konangc87) [reply src/dbsync/src/sqlite_wrapper.cpp:99]: if rollback fails there not much left to do and destructors must not throw exceptions.

File: src/dbsync/src/sqlite_wrapper.h (hunk 2 above)
Comment (konangc87) [reply src/dbsync/src/sqlite_wrapper.h:76]: will do

File: src/dbsync/src/sqlite_wrapper.h (hunk 3 above)
Comment (konangc87) [reply src/dbsync/src/sqlite_wrapper.h:78]: no, will remove it.

**Comment by konangc87** [comment]
Valgrind on tests:
image
UT coverage:
image

**Pull request merged**